modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/IgniteMath.java

```diff
@@ -308,17 +297,20 @@ public static short convertToShortExact(long x) {
 
     /** Cast value to {@code short}, throwing an exception if the result overflows an {@code short}. */
     public static short convertToShortExact(double x) {
-        if (x > Short.MAX_VALUE || x < Short.MIN_VALUE)
```

Comment:
```
x = round(x);
```
And left the old code?
In this case we don't need arrays SHORT_BOUNDS and other.

Reply:
The overflow exception might be thrown when must not be. 32767.1 is ok. Will be casted to 32767 with the new code. But the overflow-exception is thrown with the old code.

Reply:
New code still has errors on bounds.
Condition can be changed to: `if (x >= Short.MAX_VALUE + 1d || x =< Short.MIN_VALUE - 1d)`

Reply:
New code still has errors on bounds.
Condition can be changed to: `if (x >= Short.MAX_VALUE + 1d || x =< Short.MIN_VALUE - 1d)`

Reply:
Was fixed. Lost in commits. Pushed.